{{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %}

```diff
@@ -1,62 +1,145 @@
 package {{cookiecutter.top_level_domain_name}}.{{cookiecutter.second_level_domain_name}}.{{cookiecutter.project_package_name}}.security;
 
-import static {{cookiecutter.top_level_domain_name}}.{{cookiecutter.second_level_domain_name}}.{{cookiecutter.project_package_name}}.security.SecurityConstants.HEADER_STRING;
-import static {{cookiecutter.top_level_domain_name}}.{{cookiecutter.second_level_domain_name}}.{{cookiecutter.project_package_name}}.security.SecurityConstants.SECRET;
-import static {{cookiecutter.top_level_domain_name}}.{{cookiecutter.second_level_domain_name}}.{{cookiecutter.project_package_name}}.security.SecurityConstants.TOKEN_PREFIX;
-
-import java.io.IOException;
-import java.util.ArrayList;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+{%- if cookiecutter.has_lombok == "y" %}
+import lombok.extern.slf4j.Slf4j;
+{%- endif %}
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
 
-import com.auth0.jwt.JWT;
-import com.auth0.jwt.algorithms.Algorithm;
+/**
+ * <h1>JWT Authorization Filter</h1>
+ *
+ * <p>This implementation supports JWT authentication. It is an authentication
+ * filter to issue JWTs to users sending credentials</p>
+ * 
+ * @author crowdbotics.com
+ */
+{%- if cookiecutter.has_lombok == "y" %}
+@Slf4j
+{%- endif %}
+public class JWTAuthorizationFilter 
+	extends BasicAuthenticationFilter 
+{
+	/**
+	 * Autowired constructor for {@link JWTAuthorizationFilter}.
+	 *
+	 * @param _authenticationManager		{@link AuthenticationManager}
```

Comment:
Let's replace tabs between authenticationManager and {@link... with a space

Reply:
Done

Reply:
Do want the same for JWTAuthenticationFilter.java?

Reply:
Sure. I see the same in getAuthentication() as well.